### PR TITLE
email_notification 

### PR DESCRIPTION
**Pull request by Rohithknaidu**

### CODE REVIEW

File: email.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	"gopkg.in/mail.v2"
+)
+
+func notify(t trackInput) error {
+	m := mail.NewMessage()
+	m.SetHeader("From", "smulytestground@gmail.com")
+	m.SetHeader("To", "msrinivasareddy@gmail.com")
+	if t.TypeOfRequest=="AVAILABILITY"{
+	m.SetHeader("Subject", "Availability update Notification")
```
Comment (muly) [inline email.go:16]: format the code

File: email.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	"os"
+
+	"gopkg.in/mail.v2"
+)
+
+func notify(t trackInput) error {
+	m := mail.NewMessage()
+	m.SetHeader("From", "smulytestground@gmail.com")
+	m.SetHeader("To", "msrinivasareddy@gmail.com")
+	if t.TypeOfRequest=="AVAILABILITY"{
```
Comment (muly) [inline email.go:15]: replace the request type with constants.

File: email.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"gopkg.in/mail.v2"
+)
+
+func notify(t trackInput) error {
```
Comment (muly) [inline email.go:11]: let's separate the email preparation from the actual send email part.

File: email.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	m.SetHeader("To", "msrinivasareddy@gmail.com")
+	if t.TypeOfRequest=="AVAILABILITY"{
+	m.SetHeader("Subject", "Availability update Notification")
+	m.SetBody("text/plain", "product %s is available"+t.Url)
+	}else if t.TypeOfRequest=="PRICE"{
+		m.SetHeader("Subject", "price update Notification")
+		m.SetBody("text/plain", "product %s is available with the minimum cost you needed"+t.Url)
+	}
+	password := os.Getenv("GMAIL_PASSWORD")
+	d := mail.NewDialer("smtp.gmail.com", 587, "smulytestground@gmail.com", password)
```
Comment (muly) [inline email.go:23]: let's make the dialer object package level, like the `firestoreClient` object, and add a `init...()` function.

**Review by muly — CHANGES_REQUESTED** [review]

File: email.go (hunk 4 above)
Comment (Rohithknaidu) [reply email.go:23]: i didn't understand it sir

File: email.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"gopkg.in/mail.v2"
+)
+
+var emailclient *mail.Dialer
```
Comment (muly) [inline email.go:10]: use camel case

File: email.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+func initEmailClient() {
+
```
Comment (muly) [inline email.go:13]: empty space

File: email.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+	password := os.Getenv("GMAIL_PASSWORD")
+	emailclient = mail.NewDialer("smtp.gmail.com", 587, "smulytestground@gmail.com", password)
+
```
Comment (muly) [inline email.go:16]: empty space

File: email.go [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+func requesttype(*mail.Message) {
```
Comment (muly) [inline email.go:19]: need a better name for the function. also please add a comment before the function code describing the purpose of the function. to this and all the functions

File: email.go [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+	var t trackInput
+	m := mail.NewMessage()
+	m.SetHeader("From", "smulytestground@gmail.com")
+	m.SetHeader("To", "rohith.knaidu0125@gmail.com")
+	// if err!=nil{
+	// 	log.Println("error during ",err)
+	// }
+	if err := emailclient.DialAndSend(m); err != nil {
```
Comment (muly) [inline email.go:27]: the email will be sent with empty subject and body

File: util.go [hunk 10]
```diff
@@ -59,7 +61,8 @@ func shouldNotify(i trackInput, p product) bool {
 }
 
 func notify(t trackInput) error {
-	// TODO: implement later
+	
+	requesttype(&mail.Message{})
```
Comment (muly) [inline util.go:65]: mock notification log is no longer necessary

File: email.go [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+func initEmailClient() {
+
+	password := os.Getenv("GMAIL_PASSWORD")
+	emailclient = mail.NewDialer("smtp.gmail.com", 587, "smulytestground@gmail.com", password)
+
+}
+
+func requesttype(*mail.Message) {
+	var t trackInput
```
Comment (muly) [inline email.go:20]: t is empty

**Review by muly — CHANGES_REQUESTED** [review]

File: email.go (hunk 8 above)
Comment (Rohithknaidu) [reply email.go:19]: i tried

**Review by muly — APPROVED** [review]

**Pull request merged**